docs/differences-from-v1.md

```diff
@@ -0,0 +1,69 @@
+  - ["Modules" are now "analysis targets"](#modules-are-now-analysis-targets)
+  - [Automatic analysis target discovery](#automatic-analysis-target-discovery)
+  - [Automatic analysis target strategy selection](#automatic-analysis-target-strategy-selection)
+  - [Improved correctness](#improved-correctness)
+  - [Improved debug logging](#improved-debug-logging)
+- [How to upgrade to FOSSA 2.x](#how-to-upgrade-to-fossa-2x)
+  - [Remove calls to `fossa init`](#remove-calls-to-fossa-init)
+  - [Migrate your configuration file](#migrate-your-configuration-file)
+  - [Migrate "archive upload" targets](#migrate-archive-upload-targets)
+
+## What's new in FOSSA 2.x?
+
+Our focus in CLI 2.x is making FOSSA integrations easier to deploy by reducing the amount of configuration needed by users. Our goal is to get as close to turnkey deployment on as many build systems and codebases as possible.
+
+### "Modules" are now "analysis targets"
+
+In 1.x, the CLI was configured by "modules", which were entrypoints to user programs that the CLI should analyze.
+
+In 2.x, we've renamed "modules" to "analysis targets", and simplified how analysis targets are defined.
+
+We renamed analysis targets to avoid confusing naming conflicts with existing language-specific "modules". Analysis targets are no longer defined by multiple, confusingly similar fields (e.g. "Dir" vs. "Path"), and are now simple `(type, target)` tuples.
+
+### Automatic analysis target discovery
+
+FOSSA CLI 2.x now does automatic analysis target discovery when you run `fossa analyze`.
+
+This means that `fossa init` is no longer required. `fossa init` is now a no-op that emits a warning, and will be removed in a future release.
+
+### Automatic analysis target strategy selection
+
+In 1.x, modules were manually configured with "strategies" that specified _how_ a module should be analyzed. While `fossa init` attempted to choose the best strategy, manual intervention was sometimes required depending on the project's setup.
+
+In 2.x, the CLI now automatically selects the optimal strategy for analysis targets given the current environment (e.g. whether a build tool is available). This is possible because discovery and analysis are now one step, so we can check the suitability of analysis strategies while discovering targets.
+
+### Improved correctness
+
+Analysis strategies in CLI 2.x have substantive improvements in correctness and reliability across all language integrations. 2.x has much stronger compile-time correctness guarantees in its parsers, and uses build tool plugins instead of output parsing for target types where output parsing was brittle.
+
+### Improved debug logging
+
+CLI 2.x has much better debug logging, including a new feature called "replay logging" which allows developers to perfectly reproduce a bug report given a replay log. This is made possible by stronger compile-time guarantees that ensure all effects that occur during analysis are logged for replay.
```

Comment:
Should provide an example on how to use replay logging for reporting bugs.

Reply:
see replay logging docs in the `devdocs` folder

Reply:
I'm going to skip on this for now, because I think we need to improve our debugging experience in general (see https://github.com/fossas/team-analysis/issues/610), and I suspect this will quickly become outdated.